lib/ansible/modules/system/sysctl.py

```diff
@@ -386,9 +386,21 @@ def main():
             ignoreerrors = dict(default=False, type='bool'),
             sysctl_file = dict(default='/etc/sysctl.conf', type='path')
         ),
-        supports_check_mode=True
+        supports_check_mode=True,
+        required_if=[('state', 'present', ['value'])],
     )
 
+    if module.params['name'] is None:
+        module.fail_json(msg="name can not be None")
+    if module.params['state'] == 'present' and module.params['value'] is None:
+        module.fail_json(msg="value can not be None")
```

Comment:
Same here, if the parameter is required, it cannot be None. That's implicit, so this check should not be necessary.